### PR TITLE
Add support for Tales of Xillia

### PR DESCRIPTION
**Pull request by AphelionWasTaken**
Adds support for the NA versions of Tales of Xillia.

Allows for camera movement during skits, which isn't an issue. The camera still properly locks itself in place during combat and cutscenes and you regain control afterwards.

Sorry for the mess; everything can be squashed into one commit on merge... I will make sure to keep things clean in the future.

### CODE REVIEW

File: Xillia1.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        public override void UpdateCamera(int diffX, int diffY)
+        {
+            m_camera.Vert = IPCUtils.ReadFloat(m_ipc, m_address + 4);
+            m_camera.Hor = IPCUtils.ReadFloat(m_ipc, m_address);
+            m_camera.Update(diffX * SensModifier, diffY * SensModifier);
+            IPCUtils.WriteFloat(m_ipc, m_address + 4, m_camera.Vert);
+            IPCUtils.WriteFloat(m_ipc, m_address, m_camera.Hor);
+        }
+    }
+}
```
Comment (isJuhn) [inline Xillia1.cs:23]: missing newline

File: Xillia1.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
+using System;
+using KAMI.Cameras;
+using KAMI.Utilities;
+
+namespace KAMI.Games
+{
+    class Xillia1 : Game<HAVACamera>
+    {
+
```
Comment (n-laszlo-m) [inline Xillia1.cs:9]: please remove this empty line

File: Xillia1.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        {
+            if (DerefChain.VerifyChains(m_hor, m_vert))
+            {
+            m_camera.Hor = (float)(IPCUtils.ReadFloat(m_ipc, (uint)m_hor.Value));
+            m_camera.Vert = (float)(IPCUtils.ReadFloat(m_ipc, (uint)m_vert.Value));
+            
+            m_camera.Update(diffX * SensModifier, diffY * SensModifier);
+
+            IPCUtils.WriteFloat(m_ipc, (uint)m_hor.Value, m_camera.Hor);
+            IPCUtils.WriteFloat(m_ipc, (uint)m_vert.Value, m_camera.Vert);
```
Comment (n-laszlo-m) [inline Xillia1.cs:32]: please indent this

File: Xillia1.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
```
Comment (n-laszlo-m) [inline Xillia1.cs:33]: please remove newline here as well

**Review by n-laszlo-m — CHANGES_REQUESTED** [review]
Could you also add the game to the README.md compat list (following the already present format)? And maybe also fix the typo I made there with `Ratched & Clank: Tools of Destruction`? :P

**Review by dio-gh — CHANGES_REQUESTED** [review]
Looks pretty solid! Could you please also adjust readme.md and add the game to the quasi-compatlist following the format already laid out? Also, could you please fix my typo in r&c:tod while there? 😅 Accidentally wrote Ratched instead of Ratchet.

**Review by dio-gh — APPROVED** [review]
LGTM!

File: KAMI/Games/Xillia1.cs [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+using KAMI.Utilities;
+
+namespace KAMI.Games
+{
+    class Xillia1 : Game<HAVACamera>
+    {
+        const uint BaseAddress = 0xE1552C;
+
+        DerefChain m_hor;
+        DerefChain m_vert;
```
Comment (isJuhn) [inline KAMI/Games/Xillia1.cs:12]: You can use a single `DerefChain` and then when reading do
```
                m_camera.Hor = IPCUtils.ReadFloat(m_ipc, (uint)m_addr.Value);
                m_camera.Vert = IPCUtils.ReadFloat(m_ipc, (uint)(m_addr.Value + 4));
```

File: KAMI/Games/Xillia1.cs [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+            m_hor = baseChain.Chain(0x78);
+            m_vert = baseChain.Chain(0x7C);
+        }
+
+        public override void UpdateCamera(int diffX, int diffY)
+        {
+            if (DerefChain.VerifyChains(m_hor, m_vert))
+            {
+                m_camera.Hor = (float)(IPCUtils.ReadFloat(m_ipc, (uint)m_hor.Value));
+                m_camera.Vert = (float)(IPCUtils.ReadFloat(m_ipc, (uint)m_vert.Value));
```
Comment (isJuhn) [inline KAMI/Games/Xillia1.cs:26]: redundant cast to float

**Review by aaaaadox — COMMENTED** [review]
hm

**Pull request merged**